meteor/server/api/ingest/rundownInput.ts

```diff
@@ -2049,11 +2059,100 @@ function makeChangeObj(segmentId: SegmentId): SegmentChanges {
 		},
 	}
 }
-
 function unsyncSegmentOrRundown(cache: CacheForRundownPlaylist, rundownId: RundownId, segmentId: SegmentId) {
 	if (Settings.allowUnsyncedSegments) {
 		ServerRundownAPI.unsyncSegmentInner(cache, rundownId, segmentId)
 	} else {
 		ServerRundownAPI.unsyncRundownInner(cache, rundownId)
 	}
 }
+export function updateExpectedPackagesOnRundown(cache: CacheForRundownPlaylist, rundownId: RundownId): void {
+	check(rundownId, String)
+
+	// @todo: this call is for backwards compatibility and soon to be removed
+	updateExpectedMediaItemsOnRundown(cache, rundownId)
+
+	const rundown = cache.Rundowns.findOne(rundownId)
+	if (!rundown) {
+		cache.deferAfterSave(() => {
+			const removedItems = ExpectedPackages.remove({
+				rundownId: rundownId,
+			})
+			logger.info(`Removed ${removedItems} expected media items for deleted rundown "${rundownId}"`)
+		})
+		return
+	} else {
+		const studioId = rundown.studioId
+
+		cache.deferAfterSave(() => {
+			const pAdlibs = asyncCollectionFindFetch(AdLibPieces, { rundownId: rundown._id })
+			const pActions = asyncCollectionFindFetch(AdLibActions, { rundownId: rundown._id })
+			const pStudio = asyncCollectionFindOne(Studios, { _id: studioId })
+
+			const pieces = cache.Pieces.findFetch({
+				startRundownId: rundown._id,
+			})
+
+			const adlibs = waitForPromise(pAdlibs)
```

Comment:
baseline adlibs and actions should be included here too.

you could write this as `const [adlibs, actions, studio] = waitForPromiseAll([pAdlibs, pActions, pStudio])`